utils/gatewayUtils.js

```diff
@@ -20,31 +20,36 @@ const runGqlQuery = async ({ query, variables, jwt }) => {
   });
 };
 
-const uploadFileFromString = (fileData, fileName, gqlVariableName) => {
-  return { fileData, fileName, gqlVariableName };
+const uploadFileFromString = (fileData, fileName) => {
+  return { fileData, fileName };
 };
 
 /**
  *
- * @param options.files is an object of structure {fileData, fileName, gqlVariableName} - use stringToUploadFile() method to generate
+ * @param options.files is an object of structure {fileData, fileName } - use stringToUploadFile() method to generate
  */
-const runGqlUpload = async ({ query, variables, jwt, files }) => {
+const runGqlUpload = async ({ query, variables, jwt, files, asArray = true }) => {
   const formData = new FormData();
   const updatedVariables = { ...variables };
   const map = {};
+  updatedVariables.files = [];
 
   for (let i = 0; i < files.length; i++) {
     const file = files[i];
     const index = '' + (i + 1);
-    updatedVariables[file.gqlVariableName] = null;
-    map[index] = [`variables.${file.gqlVariableName}`];
+    if (asArray) {
+      updatedVariables.files.push(null);
+    } else {
+      updatedVariables.files = null;
+    }
+    map[index] = asArray ? [`variables.files.${i}`] : [`variables.files`];
```

Comment:
not required, but if you feel like, this is a perfect `reduce` use-case.
```
const map = files.reduce((acc, file) => accumulate(acc, file), {})
```
where `accumulate` is basically what's happening in this loop